powerlevel9k.zsh-theme

```diff
@@ -858,6 +858,16 @@ prompt_dir() {
           [[ $current_path == "~"* ]] && current_path="~${trunc_path//${home_path}/}" || current_path="/${trunc_path}"
         fi
       ;;
+      truncate_to_unique_but_last)
+        # see truncate_to_unique
+        if (( ${#current_path} > 1 )); then # root and home are exceptions and won't have paths
+          # cheating here to retain ~ as home folder
+          local home_path="$(getUniqueFolder $HOME)"
+          trunc_path="$(getUniqueFolder $(dirname $PWD))"
```

Comment:
Use `${PWD:h}` instead of `$(dirname $PWD)` - it's internal and thus a lot faster 😉

Reply:
Ah, good to know, thanks.
Using `:h` I found a way to use substitution to get the left path so I can split it up more easily. (`:t` would not work in this case as I understand it)

This also does not break if there are more `:h`s than directory levels.
```
% unique_part="${PWD:h:h}"; echo $unique_part ${PWD#${unique_part}}
/home/chkugler/git/private/confrc /submodules/powerlevel9k
% unique_part="${PWD:h:h:h:h:h:h:h:h:h:h:h:h:h:h:h:h:h:h}"; echo $unique_part ${PWD#${unique_part}} 
/ home/chkugler/git/private/confrc/submodules/powerlevel9k
```

I am having trouble getting a variable to control how many directories are cut off though
```
% echo $PWD
/home/user/git/private/confrc/submodules/powerlevel9k
% LEN=3
% # 3 for PWD, 2 characters in ":h", $LEN for setable variable 
% local_pwd=${(r#$((3+2*${LEN}))##:h#)p}                       
% echo $local_pwd                  
PWD:h:h:h
% echo ${PWD:h:h:h}                
/home/user/git/private
% echo ${(P)local_pwd}
/home/user/git/private/confrc/submodules/powerlevel9k
```
I suspect `(P)` somehow ignores my `:h`s. Any idea on that?


On a side note: the version in this pull request breaks stuff if your in a root folder, e.g. `/etc`, `/home`, `/tmp`. I'll try to fix this and add it to the tests if I can.

Reply:
From [ZSH Expansion](http://zsh.sourceforge.net/Doc/Release/Expansion.html)

`21. Re-evaluation
Any ‘(e)’ flag is applied to the value, forcing it to be re-examined for new parameter substitutions, but also for command and arithmetic substitutions.`

So try `echo ${(e)local_pwd}` instead